modules/gcp/gke/variables.tf

```diff
@@ -0,0 +1,197 @@
+  type        = string
+  description = "The _name_ of the secondary subnet to use for services"
+}
+
+variable "subnetwork_cidr" {
+  description = "Cidr of the subnetwork"
+  type        = string
+}
+
+variable "pod_subnet_cidr_block" {
+  type        = string
+  description = "Pod subnetwork cidr block"
+}
+
+variable "service_subnet_cidr_block" {
+  type        = string
+  description = "Services subnetwork cidr block"
+}
+
+##The variable value is important if you would want to create a private cluster
+###################
+# Variables for Private Cluster
+####################
+variable "enable_private_endpoint" {
+  type        = bool
+  description = "Kubernetes cluster endpoint get private ip if you pass it as true"
+  default     = false
+}
+
+variable "enable_private_nodes" {
+  type        = bool
+  description = "Nodes inside the nodepool get's privateip only when we pass is as true. Since we are using private subnets we want all of our nodes to have priavate IP"
+  default     = false
+}
+
+/* Note: When you Pass "enable_private_endpoint" to true you must pass the value for below variable */
+variable "master_authorized_networks" {
+  type        = list(object({ cidr_block = string, display_name = string }))
+  description = "List of master authorized networks. If none are provided, disallow external access (except the cluster node IPs, which GKE automatically whitelists)."
+  default     = []
+}
+
+/* CIDR Block will be used by GKE and creats endpoint and peer with our network */
+variable "master_ipv4_cidr_block" {
+  type        = string
+  description = "(Beta) The IP range in CIDR notation to use for the hosted master network"
+  default     = "10.0.0.0/28"
+}
+
+
+/* Addons */
+variable "http_load_balancing" {
+  type        = bool
+  description = "Enable httpload balancer addon"
+  default     = false
+}
+
+
+variable "filestore_csi_driver" {
+  type        = bool
+  description = "The status of the Filestore CSI driver addon, which allows the usage of filestore instance as volumes"
+  default     = true
+}
+
+variable "enable_vertical_pod_autoscaling" {
+  type        = bool
+  description = "Vertical Pod Autoscaling automatically adjusts the resources of pods controlled by it"
+  default     = false
+}
+
+variable "horizontal_pod_autoscaling" {
+  type        = bool
+  description = "Enable horizontal pod autoscaling addon"
+  default     = true
+}
+
+
+variable "kubernetes_version" {
+  type        = string
+  description = "The Kubernetes version of the masters."
+  default     = "latest"
+}
+
+variable "initial_node_count" {
+  type        = number
+  description = "The number of nodes to create in this cluster's default node pool."
+  default     = 1
+}
+
+variable "cluster_resource_labels" {
+  type        = map(string)
+  description = "The GCE resource labels (a map of key/value pairs) to be applied to the cluster"
+  default     = {}
+}
+
+variable "node_pools" {
+  type        = list(map(string))
```

Comment:
Please have correct type defs wherever possible. `map(string)` does not specify the list of fields that need to be passed